libs/fslock/lock_unix.go

```diff
@@ -1,6 +1,3 @@
-//go:build darwin || freebsd || linux
```

Comment:
we still need this build directive

Reply:
Sorry I blindly removed this b/c I thought both build directives were no longer needed. I looked at the previous commits:
1. https://github.com/rootulp/celestia-node/commit/b15571ca1bfb3264d4e4861a67e48aad68502016
2. https://github.com/rootulp/celestia-node/commit/1d807a4aa171604ba18732a75972ed14d18dcfa8

and learned that `//go:build` is the preferred build directive for [Go 1.17+](https://go.dev/doc/go1.17#language): 

> The go command now understands //go:build lines and prefers them over // +build lines. 

---

Based on [this article](https://www.digitalocean.com/community/tutorials/building-go-applications-for-different-operating-systems-and-architectures) I think the `_unix.go` file suffix isn't sufficient to inform the Go compiler to only build this file for Unix operating systems:

> Because unix is not a valid GOOS, the _unix.go suffix has no meaning to the Go compiler. It does, however, convey the intended purpose of the file. Like the os/path_unix.go file, your path_unix.go file still needs to use build tags, so keep that file unchanged.

so I think you are right and we still need one of these. I will add it back.

Reply:
Thank you ❤️